pkg/services/mysqldb/provision.go

```diff
@@ -30,6 +32,39 @@ func (s *serviceManager) ValidateProvisioningParameters(
 			fmt.Sprintf(`invalid sslEnforcement option: "%s"`, pp.SSLEnforcement),
 		)
 	}
+
+	startIP := net.ParseIP(pp.FirewallIPStart)
+	if pp.FirewallIPStart != "" && net.ParseIP(pp.FirewallIPStart) == nil {
+		return service.NewValidationError(
+			"firewallStartIpAddress",
+			fmt.Sprintf(`invalid firewallStartIpAddress option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+
+	endIP := net.ParseIP(pp.FirewallIPEnd)
+	if pp.FirewallIPEnd != "" && net.ParseIP(pp.FirewallIPEnd) == nil {
+		return service.NewValidationError(
+			"firewallEndIpAddress",
+			fmt.Sprintf(`invalid firewallEndIpAddress option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+
+	if startIP != nil && endIP == nil {
+		return service.NewValidationError(
+			"firewallEndIpAddress",
+			fmt.Sprintf(`invalid firewallEndIpAddress option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+
+	startBytes := startIP.To4()
+	endBytes := endIP.To4()
+	if bytes.Compare(startBytes, endBytes) > 0 {
```

Comment:
This is a clever trick. Can you just add a comment to explain how it works?

Reply:
Yep, will do.